lib/cufft/fft.jl

```diff
@@ -25,32 +25,34 @@ Base.:(*)(p::ScaledPlan, x::DenseCuArray) = rmul!(p.p * x, p.scale)
 
 # N is the number of dimensions
 
-mutable struct CuFFTPlan{T<:cufftNumber,S<:cufftNumber,K,inplace,N} <: Plan{S}
+mutable struct CuFFTPlan{T<:cufftNumber,S<:cufftNumber,K,inplace,N,R,B} <: Plan{S}
```

Comment:
Yes, and I also take this as an opportunity to have a concrete type fo `region` in the structure.

Reply:
I trust you have good reasons for this, but it's not always advantageous to do so. The CUTENSOR wrappers, for example, cause excessive specialization for little gain, since these calls are very coarse-grained anyway (yet leading to very long test times). In this case, though, the specialization is already there (since `B` doesn't introduced new information already contained in `inplace` and `T`).

Reply:
I haven't changed the dispatch for the FFT routines, so I didn't expect any slowdown. I simply made the low-level structure fully inferable.  
I'll run a quick regression test for `fft`/`ifft` before and after this PR to check if there's any noticeable difference.

I didn't test it without the type `B` in the structure, but I observed a nice speed-up in my code when adding the buffer inside for `rfft`/`irfft`.  
Julia also stopped crashing due to OOM issues.